### PR TITLE
add PerformanceProjectAction only to runs, which contain PerformanceBuildAction

### PR DESCRIPTION
**Pull request by rpionke**
The PerformanceProjectAction gets added to every pipeline run. This Performance trend is always empty:
image

This PR simplifies the WorkflowActionsFactory, so that only runs containing the PerformanceBuildAction are getting the PerformanceProjectAction added.
Furthermore i used 'mvn tidy:pom' to sort the pom.xml and increment some workflow dependencies.

### CODE REVIEW

**Comment by undera** [comment]
Looks like it broke unit tests...

**Comment by rpionke** [comment]
yeah, i forgot to fix the unit test. this test shows that every pipeline job gets the PerformanceProjectAction

File: src/test/java/hudson/plugins/performance/workflow/WorkflowActionsFactoryTest.java [hunk 1]
```diff
@@ -39,6 +41,33 @@ public void evaluate() throws Throwable {
                 r = p.scheduleBuild2(1).waitForStart();
                 story.j.assertBuildStatus(Result.SUCCESS, story.j.waitForCompletion(r));
                 Collection<? extends Action> actions = factory.createFor(p);
+                assertEquals(0, actions.size());
+            }
+        });
+    }
+
+    @Test
+    public void testFlowWithProjectAction() throws Exception {
```
Comment (rpionke) [inline src/test/java/hudson/plugins/performance/workflow/WorkflowActionsFactoryTest.java]: mhhhhn thats not true: https://github.com/rpionke/performance-plugin/blob/f055afc0602556f2cece6e1656f0a85cea8ee64e/src/test/java/hudson/plugins/performance/workflow/WorkflowActionsFactoryTest.java#L71-L72

Comment (undera) [inline src/test/java/hudson/plugins/performance/workflow/WorkflowActionsFactoryTest.java]: @codacy-bot codacy is not perfect sometimes, forgive it please

**Pull request merged**